src/libraries/System.IO.MemoryMappedFiles/tests/MemoryMappedFile.CreateFromFile.Tests.cs

```diff
@@ -696,13 +697,13 @@ private void WriteToReadOnlyFile(MemoryMappedFileAccess access, bool succeeds)
         public void WriteToReadOnlyFile_ReadWrite(MemoryMappedFileAccess access)
         {
             WriteToReadOnlyFile(access, access == MemoryMappedFileAccess.Read ||
-                            (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows) && geteuid() == 0));
+                            (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows) && PlatformDetection.IsSuperUser));
```

Comment:
Is it fine to remove `!RuntimeInformation.IsOSPlatform(OSPlatform.Windows) && ` since it feels redundant to check for windows twice when `IsBrowser` is false? Would hitting PNSE on windows be a problem?

Reply:
Given the way that `IsSuperUser` is written, if we remove the `Windows` check here, it would throw `PlatformNotSupportedException` and crash the tests.

Reply:
no we need to keep this since `PlatformDetection.IsSuperUser` would throw `PlatformNotSupportedException` if we ran it on Windows